applications/matter_weather_station/src/app_task.cpp

```diff
@@ -229,35 +177,13 @@ CHIP_ERROR AppTask::Init()
 		&sIdentifyTimer, [](k_timer *) { TaskExecutor::PostTask([] { IdentifyTimerHandler(); }); }, nullptr);
 	k_timer_start(&sMeasurementsTimer, K_MSEC(kMeasurementsIntervalMs), K_MSEC(kMeasurementsIntervalMs));
 
-	/* Initialize CHIP server */
-	static chip::CommonCaseDeviceServerInitParams initParams;
-	static OTATestEventTriggerDelegate testEventTriggerDelegate{ ByteSpan(sTestEventTriggerEnableKey) };
-	(void)initParams.InitializeStaticResourcesBeforeServerInit();
-
-	initParams.testEventTriggerDelegate = &testEventTriggerDelegate;
-	ReturnErrorOnFailure(chip::Server::GetInstance().Init(initParams));
-
-	gExampleDeviceInfoProvider.SetStorageDelegate(&Server::GetInstance().GetPersistentStorage());
-	chip::DeviceLayer::SetDeviceInfoProvider(&gExampleDeviceInfoProvider);
-	AppFabricTableDelegate::Init();
-
-	ConfigurationMgr().LogDeviceConfig();
-	PrintOnboardingCodes(chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE));
-
-	/*
-	 * Add CHIP event handler and start CHIP thread.
-	 * Note that all the initialization code should happen prior to this point
-	 * to avoid data races between the main and the CHIP threads.
-	 */
-	PlatformMgr().AddEventHandler(ChipEventHandler, 0);
+#ifdef CONFIG_MCUMGR_TRANSPORT_BT
+	GetDFUOverSMP().StartServer();
+#endif
 
-	err = PlatformMgr().StartEventLoopTask();
-	if (err != CHIP_NO_ERROR) {
-		LOG_ERR("PlatformMgr().StartEventLoopTask() failed");
-		return err;
-	}
+	ReturnErrorOnFailure(Nordic::Matter::StartChipServer());
 
-	return CHIP_NO_ERROR;
+	return Nordic::Matter::WaitForReadiness();
```

Comment:
Hmm, I don't get the WaitForReadiness and that InitGuard functionality. First of all, why does a user need to think about that and use it in their application? It is a bit confusing.

Reply:
The `InitGuard` object is just an utility which does the cleanup automatically, i.e. it unblocks the waiting thread on each failure can happen in the `DoInitChipServer`. Otherwise we need to call `k_condvar_signal` each time `DoInitChipServer` returns prematurely.